### PR TITLE
Apply peer authentication policy at port level

### PR DESCRIPTION
**Pull request by diemtvu**
Issue: https://github.com/istio/istio/issues/20746

### CODE REVIEW

File: pilot/pkg/security/authn/v1beta1/policy_applier.go [hunk 1]
```diff
@@ -309,6 +309,19 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 	}
 }
 
+func (a *v1beta1PolicyApplier) getMutualTLSModeForPort(endpointPort uint32) model.MutualTLSMode {
+	if a.consolidatedPeerPolicy == nil {
+		return model.MTLSPermissive
+	}
+	if a.consolidatedPeerPolicy.PortLevelMtls != nil {
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier.go]: What's the expected behavior for passthrough filter chain that doesn't have the port?

Comment (diemtvu) [reply pilot/pkg/security/authn/v1beta1/policy_applier.go]: The behavior should use the workload level mTLS (i.e a.consolidatedPeerPolicy.Mtls). Implementation-wise, you can change this function to treat a special value (i.e `0`) to ignore looking up portLevel settings.

**Pull request merged**